sleap/gui/commands.py

```diff
@@ -3470,13 +3481,205 @@ def do_action(context: CommandContext, params: dict):
         if rls is not None:
             context.openWebsite(rls.url)
 
+class ExportClipVideo(AppCommand):
+    @staticmethod
+    def do_action(context: CommandContext, params: dict):
+        """
+        Exports a pruned video clip and labels to a specified file based on selected frame range.
 
-def copy_to_clipboard(text: str):
-    """Copy a string to the system clipboard.
+        Args:
+            context (CommandContext): Contains state information like video and labels.
+            params (dict): Parameters including filename, fps, and open_when_done.
+        """
+        # Extract video and labels from context
+        video = context.state["video"]
+        labels = context.state["labels"]
+
+        # Ensure frame range is set; default to all frames if None
+        frame_range = context.state.get("frame_range", (0, video.frames))
+        # Validate frame range
+        if frame_range[0] < 0 or frame_range[1] > video.frames:
+            raise ValueError(f"Frame range {frame_range} is outside video bounds [0, {video.frames}]")
+        # Check if clip is selected, raise error if no clip selected
+        if frame_range == (0, video.frames) or frame_range == (0, 1) or frame_range[0] == frame_range[1]:
+            raise ValueError("No valid clip frame range selected! Please select a valid frame range using shift + click in the GUI.")
+
+        # Extract only the selected frames into a new Labels object
+        pruned_labels = labels.extract(
+            inds=range(*frame_range),
```

Comment:
When `inds` is either a `list` or `range`, then we just call `Labels.get` for each item in the `list` or `range` (`int` in our case). 

This means that we try to grab the item from the `Labels.labels: List[LabeledFrame]`  at the index we provided - which is not the same as grabbing the item at the frame index.

Reply:
https://github.com/talmolab/sleap/blob/1eff33dbd9e342142092d0581f79e21560ecdf2c/sleap/io/dataset.py#L779-L798
https://github.com/talmolab/sleap/blob/1eff33dbd9e342142092d0581f79e21560ecdf2c/sleap/io/dataset.py#L636-L674
https://github.com/talmolab/sleap/blob/1eff33dbd9e342142092d0581f79e21560ecdf2c/sleap/io/dataset.py#L768-L769
https://github.com/talmolab/sleap/blob/1eff33dbd9e342142092d0581f79e21560ecdf2c/sleap/io/dataset.py#L738-L739
https://github.com/talmolab/sleap/blob/1eff33dbd9e342142092d0581f79e21560ecdf2c/sleap/io/dataset.py#L552-L555
https://github.com/talmolab/sleap/blob/1eff33dbd9e342142092d0581f79e21560ecdf2c/sleap/io/dataset.py#L416

Reply:
Ok, I think I have addressed this suggestion by creating a frame_to_index mapping and then feeding in valid frame indices on line 3666-3677 in command.py. 

`# Map frame indices to the actual labeled frame objects
        frame_to_index = {lf.frame_idx: idx for idx, lf in enumerate(labels.labeled_frames) if lf.video == video}
        valid_frame_indices = [frame for frame in range(*frame_range) if frame in frame_to_index]

        # Extract only the selected frames into a new Labels object
        pruned_labels = labels.extract(
            inds=[frame_to_index[frame] for frame in valid_frame_indices],
            copy=True  # Ensures a deep copy of the extracted labels
        )`

This should now be robust to multiple videos and sparsely labelled videos.